saleor/warehouse/management.py

```diff
@@ -370,9 +408,14 @@ def get_order_lines_with_track_inventory(
 
 
 @traced_atomic_transaction()
-def deallocate_stock_for_order(order: "Order"):
+def deallocate_stock_for_order(order: "Order", manager: PluginsManager):
     """Remove all allocations for given order."""
     allocations = Allocation.objects.filter(
         order_line__order=order, quantity_allocated__gt=0
     ).select_for_update(of=("self",))
+
+    for allocation in allocations:
+        if allocation.stock.available_quantity() <= 0:
```

Comment:
We execute database queries for each allocation separately. We should do it in a single database query.